presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/RpmInstallationManager.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.agent;
+
+import io.airlift.log.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+import static com.teradata.prestomanager.agent.AgentFileUtils.downloadFile;
+import static com.teradata.prestomanager.agent.CommandExecutor.execCommandResult;
+import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.addTpchCatalog;
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.deployConfigFiles;
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.storeConfigFiles;
+import static java.lang.String.format;
+
+public class RpmInstallationManager
+        extends InstallationManager
+{
+    private static final Logger LOGGER = Logger.get(RpmInstallationManager.class);
+
+    public void installAsync(URL packageUrl, boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        File tempFile = getRpmPackage(packageUrl);
+        try {
+            int installRpm;
+            if (checkDependencies) {
+                installRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-iv", tempFile.toString());
```

Comment:
This can be modified in the future. I don't think this would reduce any code lines though.